cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -338,17 +329,18 @@ private ScanApplicationsRequest getScanRequest(String namespaceId, String artifa
   }
 
   /**
-   * Returns the info associated with the application.
+   * Returns the info associated with the latest application.
    */
   @GET
   @Path("/apps/{app-id}")
   public void getAppInfo(HttpRequest request, HttpResponder responder,
                          @PathParam("namespace-id") final String namespaceId,
-                         @PathParam("app-id") final String appId)
-    throws Exception {
-
-    ApplicationId applicationId = validateApplicationId(namespaceId, appId);
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(applicationLifecycleService.getAppDetail(applicationId)));
+                         @PathParam("app-id") final String appName) throws Exception {
+    // The version of the validated applicationId is ignored. We only use the method to validate the input.
+    ApplicationId applicationId = validateApplicationId(namespaceId, appName);
```

Comment:
I still strongly feel we need a separate object to pass versionless application around. It's needed in too many cases. Currently it's pretty unobvious where version is used vs not

Reply:
We does not nesessarily need it as part of API, but it's very usefull in service definitions

Reply:
As discussed offline, when version is not used like here, we should not operate with ApplicationId

Reply:
So, please change 
```
 ApplicationId validateApplicationId(@Nullable String namespace, @Nullable String appId)
```
to
```
 void validateApplicationId(@Nullable String namespace, @Nullable String appId)
```
Similar for the one that takes NamespaceId.
Only the versioned one should return something. The versioned should call this one and then additionally validate version.

Reply:
changed the method signature for validateApplicationId to return nothing